File: app/src/main/java/com/mobiledevpro/apptemplate/ui/mainscreen/presenter/MainPresenter.java
package com.mobiledevpro.apptemplate.ui.mainscreen.presenter;

/**
 * Presenter for main screen
 * <p>
 * Created by Dmitriy V. Chernysh on 20.10.17.
 * dmitriy.chernysh@gmail.com
 * <p>
 * https://fb.me/mobiledevpro/
 * <p>
 * #MobileDevPro
 */

public class MainPresenter implements IMain.Presenter {

    private IMain.View mView;

    @Override
    public void bindView(IMain.View view) {
        mView = view;
    }

    @Override
    public void unbindView() {
        mView = null;
    }
}

File: app/src/main/java/com/mobiledevpro/apptemplate/ui/mainscreen/presenter/MainPresenter.kt
package com.mobiledevpro.apptemplate.ui.mainscreen.presenter

import android.util.Log
import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleOwner
import com.mobiledevpro.data.LOG_TAG_DEBUG

/**
 * Presenter for main screen
 *
 * Created by Dmitriy V. Chernysh
 * dmitriy.chernysh@gmail.com
 *
 * https://instagr.am/mobiledevpro/
 *
 * #MobileDevPro
 */

class MainPresenter
constructor(private val view: IMain.View) : IMain.Presenter {

    override fun onStartView() {
        showMessage()
    }

    override fun onStopView() {

    }

    override fun onLifecycleEventTest(source: LifecycleOwner, event: Lifecycle.Event) {
        Log.d(LOG_TAG_DEBUG, "LIFE EVENT: ${event.name}")
    }

    private fun showMessage() {
/*
        Handler().postDelayed({
            view.showToast(R.string.app_name)
        }, 5000)
*/
    }
}
